strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/domain/ArtifactGroup.java

```diff
@@ -0,0 +1,74 @@
+package org.carlspring.strongbox.domain;
+
+import org.carlspring.strongbox.data.domain.GenericEntity;
+
+import javax.annotation.Nonnull;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.OneToMany;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@MappedSuperclass
+public abstract class ArtifactGroup
+        extends GenericEntity
+{
+
+    @OneToMany
+    private Set<ArtifactEntry> artifactEntries;
+
+    /**
+     * Should be unique per type
+     */
+    private String name;
+
+    private String type;
+
+    public ArtifactGroup()
+    {
+        type = getClass().getName();
```

Comment:
do we really need this? `OrientDB` have class metadata by default

Reply:
Without it, I won't be able to use [this](https://github.com/strongbox/strongbox/blob/master/strongbox-data-service/src/main/java/org/carlspring/strongbox/data/service/CommonCrudService.java#L273) method. I wish there could be only one `ArtifactGroupServiceImpl` (without the need to extend services per group) with single `ArtifactGroupServiceImpl#getEntityClass` returning `ArtifactGroup.class`

Reply:
seems I see what you mean, it probably [this](https://github.com/strongbox/strongbox/pull/987/files#diff-6af8485da2f1ab69413f1f04f4fdb475R85) place :)
anyway there is not a big deal to make a custom query `String.format("select * from %s where name=:name", artifactGroupClass)`.
but I think that we probably need to make `ArtifactGroup.name` attribute unique, so you can always use just `ArtifactGroup` class when selecting by `name`.

Reply:
> but I think that we probably need to make ArtifactGroup.name attribute unique, 

It is already made unique by `type` and `name`, see [here](https://github.com/strongbox/strongbox-db/pull/5/files#diff-9eb19f9d18f36a41149aae0396429bbaR17).

> so you can always use just ArtifactGroup class when selecting by name.

I think that group name should be unique in a specific group type, but not in general.